public/app/plugins/panel/flamegraph/components/FlameGraph/dataTransform.ts

```diff
@@ -107,7 +99,7 @@ export class FlameGraphDataContainer {
     // both a backward compatibility but also to allow using a simple dataFrame without enum config. This would allow
     // users to use this panel with correct query from data sources that do not return profiles natively.
     if (enumConfig) {
-      this.labelDisplayProcessor = getEnumDisplayProcessor(theme, enumConfig);
+      this.labelDisplayProcessor = getDisplayProcessor({ field: this.labelField, theme });
```

Comment:
not confident about this change... @aocenas can you please take a look at these changes and verify if they are ok or not? 🙏

Reply:
Update - seems like issue is that field type is not being set to enum (will follow up with an issue for this in PR description)

In meantime see following resources for testing locally:

Main branch
![main](https://github.com/grafana/grafana/assets/22381771/211c38e2-84f5-4349-93ef-58363768d81b)


PR branch
![branch](https://github.com/grafana/grafana/assets/22381771/60dde700-1cc5-4f73-ad2d-7c3040c94d8e)


Debug panel file
[debug-flamegraph.txt](https://github.com/grafana/grafana/files/12114964/debug-flamegraph.txt)

Reply:
Follow up -> if https://github.com/grafana/grafana/pull/72078 lands we can remove [`line 111: this.labelField.type = FieldType.enum;`](https://github.com/grafana/grafana/pull/64179/files#diff-1b690a4f4dadf472d3264cd213a79819ed5a2a76857c1ee3355fcc8410777032R111)

Reply:
Both the backend change and this one looks good to me.